### ISSUE TITLE
Two tray icons for discord in KDE

### ISSUE THREAD

**Issue by arcooke** [crmarsh/discord-linux-bugs#126]
This is happening today.  I just installed Discord yesterday and I'm not sure if it happened then too.. but it's showing two system tray icons.  I can interact with the left one, but the right one does nothing.

![](http://i.imgur.com/IfOb1AA.png)

I'm on KDE Neon.  I believe I'm using Discord 0.0.10 if memory serves.. I don't see a way to check the version, but I got latest yesterday.

**Comment by ahjolinna** [comment]
I can't reproduce that problem on chakraOS (plasma 5.8.3) but it could be a ubuntu/deb pkgs problem.

**Comment by OvermindDL1** [comment]
I'm using KDE Neon and see only one icon.  Usually when a second icon of something appears like that it is because because it was improperly closed.  Did that happen?  Does only one appear if you restart Plasma?

**Comment by arcooke** [comment]
Yeah it's only showing one icon today.  I think I may have figured out why this happened.  

I completely forgot that earlier in the day yesterday, Discord was acting up and refusing to connect to any voice channels.  The connection info debug screen wouldn't load anything either.  I figured Discord just completely bugged out, so I exited it and reopened it, and after that everything was fine again.

I bet that icon was some kind of phantom icon left over from Discord not being able to shut down properly.. maybe it crashed as I was exiting, and then never got to do any cleanup with the tray icon (if that's a thing)

**Comment by OvermindDL1** [comment]
> I bet that icon was some kind of phantom icon left over from Discord not being able to shut down properly.. maybe it crashed as I was exiting, and then never got to do any cleanup with the tray icon (if that's a thing)

Sounds like it, it is a common bug with GTK programs.  GTK program have to call their cleanup manually at the end to unhook their global memory and such, QT programs do not need to (never seen it happen with a QT program), and I'm pretty sure discord is using GTK for whatever reason.  Consequently if a GTK program does not close properly then it leaves certain shared pipes open, including the one controlling that icon.  So it 'technically' is a GTK bug, which makes it a Discord bug, but for discord to fix it they'd have to switch to a better rendering API, which would likely be a ton of work.  ;-)

**Comment by alexandrpaliy** [comment]
KDE5/Plasma, I don't have this issue.
